### PR TITLE
Update Add Approver Modal

### PR DESCRIPTION
**Pull request by yksolanki9**
https://user-images.githubusercontent.com/39493102/133249826-8e697b28-aa8b-41de-a469-091bd4db1c8f.mp4

### CODE REVIEW

**Comment by Dimple16** [comment]
Few comments from the video
- The parent pop over should have a top and right bottom radius
- Let us set the minimum approvers displayed in the list to 3
- The parent pop over header should have the same design as the select approvers modal
 - The cancel icon should be the `fy-close` svg
 - The save button should have the `$primary` color
 - It should have a bottom border
- Verify the chips height and inner padding in Figma

File: src/theme/colors.scss [hunk 1]
```diff
@@ -31,6 +31,7 @@ $blue-2: #ebf3fc !default;
 $green-1: #24a148 !default;
 $red-1: #da1e2817 !default;
 $red-2: #b8141d;
+$red-3: #ff3366 !default;
```
Comment (Dimple16) [inline src/theme/colors.scss:34]: this is defined as `$primary` , we can use the same one

File: src/app/shared/components/fy-apporver/add-approvers-popover/approver-dialog/approver-dialog.component.scss [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+    display: flex;
+    flex-direction: column;
+    padding: 0 10px;
+  }
+
+  &--list-element {
+    padding: 18px 0;
+    margin: 0 16px;
+    border-bottom: 1px solid #e0e0e0;
+    color: #ababab;
```
Comment (Dimple16) [inline src/app/shared/components/fy-apporver/add-approvers-popover/approver-dialog/approver-dialog.component.scss:65]: use `$gray-3`

File: src/app/shared/components/fy-apporver/add-approvers-popover/approver-dialog/approver-dialog.component.scss [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+  &--container {
+    display: flex;
+    flex-direction: column;
+    padding: 0 10px;
+  }
+
+  &--list-element {
+    padding: 18px 0;
+    margin: 0 16px;
+    border-bottom: 1px solid #e0e0e0;
```
Comment (Dimple16) [inline src/app/shared/components/fy-apporver/add-approvers-popover/approver-dialog/approver-dialog.component.scss:64]: define this color in the colors.scss

File: src/app/shared/components/fy-apporver/add-approvers-popover/approver-dialog/approver-dialog.component.scss [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+.selection-modal {
+  &--title {
+    @extend %fy-select-title;
+    padding: 0;
+    margin-left: 0;
+  }
+
+  &--cta {
+    @extend %fy-select-cta;
+    color: $red-3;
```
Comment (Dimple16) [inline src/app/shared/components/fy-apporver/add-approvers-popover/approver-dialog/approver-dialog.component.scss:13]: use `$primary`

File: src/app/shared/components/fy-apporver/add-approvers-popover/add-approvers-popover.component.ts [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+})
+export class AddApproversPopoverComponent {
+
+  @Input() approverEmailsList;
+
+  @Input() id: string;
+
+  @Input() ownerEmail: string;
+
+  @Input() from;
```
Comment (Dimple16) [inline src/app/shared/components/fy-apporver/add-approvers-popover/add-approvers-popover.component.ts:25]: can we change this to type? `this.type === 'Trips'` etc

File: src/app/shared/components/fy-apporver/add-approvers-popover/add-approvers-popover.component.ts [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+      componentProps: {
+        approverEmailsList: this.approverEmailsList.slice(),
+        initialApproverList: this.selectedApproversList.slice(),
+        id: this.id,
+        from: this.from,
+        ownerEmail: this.ownerEmail,
+      },
+      mode: 'ios',
+      ...this.modalProperties.getModalDefaultProperties(),
+      backdropDismiss: false
```
Comment (Dimple16) [inline src/app/shared/components/fy-apporver/add-approvers-popover/add-approvers-popover.component.ts:55]: `backdropDismiss` is added as true by default in the properties, any reason we are disabling this?

File: src/app/shared/components/fy-apporver/add-approvers-popover/add-approvers-popover.component.ts [hunk 7]
```diff
@@ -0,0 +1,95 @@
...
+    });
+
+    await approversListModal.present();
+
+    const { data } = await approversListModal.onWillDismiss();
+
+    if (data && data.selectedApproversList) {
+      this.selectedApproversList = data.selectedApproversList;
+      this.displayValue = data.selectedApproversList.map(selectedApprover => selectedApprover.name).slice(0, 2).join(', ');
+      if(this.selectedApproversList.length > 2) {
```
Comment (Dimple16) [inline src/app/shared/components/fy-apporver/add-approvers-popover/add-approvers-popover.component.ts:65]: `this.selectedApproversList && this.selectedApproversList.length > 2`

**Review by Dimple16 — CHANGES_REQUESTED** [review]
added comments

File: src/app/shared/components/fy-approver/add-approvers-popover/approver-dialog/approver-dialog.component.ts [hunk 8]
```diff
@@ -1,105 +1,119 @@
...
-import { AdvanceRequestService } from 'src/app/core/services/advance-request.service';
+import { switchMap, map, finalize, startWith, distinctUntilChanged } from 'rxjs/operators';
+import { ModalController } from '@ionic/angular';
 import { Employee } from 'src/app/core/models/employee.model';
-import { isEqual, cloneDeep } from 'lodash';
-import { ReportService } from 'src/app/core/services/report.service';
+import { COMMA, ENTER } from '@angular/cdk/keycodes';
+import { MatChipInputEvent } from '@angular/material/chips';
+
+type approverModel = {
```
Comment (Julias0) [inline src/app/shared/components/fy-approver/add-approvers-popover/approver-dialog/approver-dialog.component.ts:11]: approverModel -> Approver

**Review by Julias0 — APPROVED** [review]

File: src/app/shared/components/fy-approver/add-approvers-popover/approver-dialog/approver-dialog.component.ts (hunk 8 above)
Comment (yksolanki9) [reply src/app/shared/components/fy-approver/add-approvers-popover/approver-dialog/approver-dialog.component.ts:11]: Done

**Pull request merged**